front/src/modules/simulationResult/components/SpeedSpaceChart/useSpeedSpaceChart.ts

```diff
@@ -0,0 +1,216 @@
+  type PostV2InfraByInfraIdPathPropertiesApiArg,
+  type SimulationPowerRestrictionRange,
+  type TrainScheduleResult,
+} from 'common/api/osrdEditoastApi';
+import { useInfraID } from 'common/osrdContext';
+import { sec2d3datetime } from 'modules/simulationResult/components/ChartHelpers/ChartHelpers';
+import {
+  ChartSynchronizerV2,
+  type ChartSynchronizerTrainData,
+} from 'modules/simulationResult/components/ChartHelpers/ChartSynchronizerV2';
+import { getSelectedTrainId } from 'reducers/osrdsimulation/selectors';
+import { isoDateWithTimezoneToSec } from 'utils/date';
+import { mmToM } from 'utils/physics';
+
+/** Prepare data needed for speedSpaceChart */
+const useSpeedSpaceChart = (
+  trainScheduleResult?: TrainScheduleResult,
+  stdcmResponse?: StdcmV2SuccessResponse
+) => {
+  const infraId = useInfraID();
+  const selectedTrainId = useSelector(getSelectedTrainId);
+
+  const [formattedPathProperties, setFormattedPathProperties] = useState<PathPropertiesFormatted>();
+  const [formattedPowerRestrictions, setFormattedPowerRestrictions] = useState<
+    SimulationPowerRestrictionRange[]
+  >([]);
+
+  const { data: rollingStock } =
+    osrdEditoastApi.endpoints.getRollingStockNameByRollingStockName.useQuery(
+      {
+        rollingStockName: trainScheduleResult?.rolling_stock_name as string,
+      },
+      { skip: !trainScheduleResult }
+    );
+
+  const { data: path } = osrdEditoastApi.endpoints.getV2TrainScheduleByIdPath.useQuery(
+    {
+      id: selectedTrainId as number,
+      infraId: infraId as number,
+    },
+    { skip: !selectedTrainId || !infraId }
+  );
+
+  const { data: trainSimulation } =
+    osrdEditoastApi.endpoints.getV2TrainScheduleByIdSimulation.useQuery(
+      { id: selectedTrainId as number, infraId: infraId as number },
+      { skip: !selectedTrainId || !infraId }
+    );
+
+  const [postPathProperties] =
+    osrdEditoastApi.endpoints.postV2InfraByInfraIdPathProperties.useMutation();
+
+  const simulation = trainSimulation || stdcmResponse?.simulation;
+  const pathfindingResult = path?.status === 'success' ? path : stdcmResponse?.path;
+  const departureTime =
+    trainScheduleResult?.id === STDCM_TRAIN_ID
+      ? stdcmResponse?.departure_time
+      : trainScheduleResult?.start_time;
+
+  // retrieve and format pathfinding properties
+  useEffect(() => {
+    const getPathProperties = async () => {
+      if (
+        infraId &&
+        trainScheduleResult &&
+        rollingStock &&
+        pathfindingResult &&
+        simulation?.status === 'success'
+      ) {
+        const pathPropertiesParams: PostV2InfraByInfraIdPathPropertiesApiArg = {
+          infraId,
+          props: ['electrifications', 'geometry', 'operational_points', 'curves', 'slopes'],
+          pathPropertiesInput: {
+            track_section_ranges: pathfindingResult.track_section_ranges,
+          },
+        };
+        const pathPropertiesResult = await postPathProperties(pathPropertiesParams).unwrap();
```

Comment:
In this useEffect i would also make a split here creating a fetch function

Reply:
I created a hook to fetch the data :+1: